### PR TITLE
Working on how to do commands

### PR DESCRIPTION
**Pull request by rmevans9**
I considered virtualizing the list then I realized that I don't care about that too much...yet. This is just trying to work out a good way to standardize the way commands look. This will be helpful when we get to a point that 3rd party libs can contribute their own commands to the timeline.

### CODE REVIEW

File: packages/reactotron-core-ui/src/timeline-command/timeline-command.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from "react"
+
+import { Timestamp } from "./timestamp"
+
+// TODO: Move this?
+export abstract class TimelineCommand {
```
[the remark this answers is not in the document]
Comment (rmevans9) [reply packages/reactotron-core-ui/src/timeline-command/timeline-command.tsx]: I like this idea... a lot. I will see about getting at least the `addTimelineCommand` API going.

[the remark this answers is not in the document]
Comment (ianoshorty) [reply packages/reactotron-core-ui/src/timeline-command/timeline-command.tsx]: This looks awesome.

**Comment by rmevans9** [comment]
Updated to start moving toward the API suggested by @skellock

File: packages/reactotron-core-ui/src/reactotron-app.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export interface TimelineCommandOptions {
+    command: string
+    type: string
+    component: any // TODO: What type for this?
+    toolbar: any // TODO: What type for this?
+    preview: any // TODO: What type for this?
+}
+
+export interface ReactotronApp {
+    addTimelineCommand: (options: TimelineCommandOptions) => ReactotronApp
```
Comment (rmevans9) [inline packages/reactotron-core-ui/src/reactotron-app.ts]: Startin small. Will eventually have stuff for tabs and such

File: packages/reactotron-core-ui/src/timeline-command/timeline-command.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  render() {
+    const { command, timelineConfig } = this.props
+    const { expanded } = this.state
+
+    const PreviewComponent = timelineConfig.preview
+    const BodyComponent = timelineConfig.component
+
+    return (
```
Comment (rmevans9) [inline packages/reactotron-core-ui/src/timeline-command/timeline-command.tsx]: Need a lot of work here to make it look right and provide all features that were in here before.

File: packages/reactotron-core-ui/src/timeline-command/timestamp.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  deltaTime?: number
+}
+
+export function Timestamp({ date, deltaTime }: Props) {
+  const hourMin = format(date, "h:mm:")
+  const secMilli = format(date, "ss.SSS")
+  const delta = deltaTime ? `+${deltaTime} ms` : ""
+
+  return (
+    <div>
```
Comment (rmevans9) [inline packages/reactotron-core-ui/src/timeline-command/timestamp.tsx]: STYLING NEEDED!

File: packages/reactotron-server/src/app/index.html [hunk 5]
```diff
@@ -10,8 +10,8 @@
 </head>
 
 <body>
-  <div id='root' class="flex flex-col min-h-screen text-default font-sans" />
+  <div id='root' class="flex flex-col min-h-screen max-h-screen text-default font-sans" />
```
Comment (rmevans9) [inline packages/reactotron-server/src/app/index.html]: Was having a odd double scroll view... needed to lock the max height too.

File: packages/reactotron-server/src/app/reactotron-app.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { ReactotronApp, TimelineCommandOptions } from "reactotron-core-ui"
+
+class Reactotron implements ReactotronApp {
```
Comment (rmevans9) [inline packages/reactotron-server/src/app/reactotron-app.ts]: I wanted the implementation in server itself so offloaded the interface off to the core-ui package and implement it as a singleton here.

File: packages/reactotron-server/src/app/system-config.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { ReactotronApp } from "reactotron-core-ui"
+import { ApiResponseTimeline, ApiResponsePreview } from "./commands/ApiResponse"
+import { SagaTaskTimeline, SagaTaskPreview } from "./commands/SagaTaskComplete"
+
+export function registerSystemTimelineCommands(app: ReactotronApp) {
+  app.addTimelineCommand({
```
Comment (rmevans9) [inline packages/reactotron-server/src/app/system-config.ts]: 🎉

File: packages/reactotron-server/src/app/system-config.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import { ReactotronApp } from "reactotron-core-ui"
+import { ApiResponseTimeline, ApiResponsePreview } from "./commands/ApiResponse"
+import { SagaTaskTimeline, SagaTaskPreview } from "./commands/SagaTaskComplete"
+
+export function registerSystemTimelineCommands(app: ReactotronApp) {
```
Comment (rmevans9) [inline packages/reactotron-server/src/app/system-config.ts]: This particular file could import the ReactotronApp directly but I wanted to emulate it more like what the plugins will end up with. One day hopefully most of the commands that end up in here will be in a plugin (looking at you `saga.task.complete`)

**Comment by skellock** [comment]
Looking great dude!

**Comment by skellock** [comment]
Would you mind if I merged this in?

**Comment by skellock** [comment]
I merged it in.  I'm pretending you said yes.

**Pull request merged**